src/vehicle_use.cpp

```diff
@@ -1092,6 +1093,38 @@ void vehicle::play_chimes()
     }
 }
 
+void vehicle::crash_terrain_around() {
+    const tripoint around[] = { {1, 1, 0}, { 0,1,0 }, {-1, 1, 0}, { 1, 0, 0}, {0, 0, 0}, { -1, 0, 0}, {1, -1, 0}, {0, -1, 0}, {-1, -1, 0} };
+    if (total_power_w() <= 0)
+        return;
+    for (const vpart_reference& vp : get_enabled_parts("CRASH_TERRAIN_AROUND")) {
+        tripoint crush_target(0, 0, -OVERMAP_LAYERS);
+        const tripoint start_pos = vp.pos();
+        const transform_terrain_data& ttd = vp.info().transform_terrain;
+        for (int i = 0; i < 9 && crush_target.z == -OVERMAP_LAYERS; i++) {
+            tripoint cur_pos = start_pos + around[i];
+            bool busy_pos = false;
+            for (const vpart_reference& vp_tmp : get_all_parts()) {
+                busy_pos |= vp_tmp.pos() == cur_pos;
+            }
+            for (const std::string& flag : ttd.pre_flags) {
+                if (g->m.has_flag(flag, cur_pos) && !busy_pos) {
+                    crush_target = cur_pos;
+                    break;
+                }
+            }
+        }
+        if (crush_target.z != -OVERMAP_LAYERS) { //target chosen
```

Comment:
`OVERMAP_LAYERS` equals to `1 + OVERMAP_DEPTH + OVERMAP_HEIGHT`. You probably need to use `-OVERMAP_DEPTH ` here.

Reply:
Here, I'm checking the value, that never could exist for understanding did I found something or not. As you can understand, any rightful value could not be below -OVERMAP_DEPTH. So, I can use  -OVERMAP_DEPTH - 1 or -(OVERMAP_DEPTH  + 1) or -(1 + OVERMAP_DEPTH + OVERMAP_HEIGHT).

Reply:
Right, I started reading code from the bottom, but now see what you mean. I think you better use something else here to avoid confusion - maybe use one of `inbounds` function that we have?

Reply:
Yes, it is better. Thanks for help ;-)